src/__tests__/authorizationMiddleware.test.ts

```diff
@@ -0,0 +1,78 @@
+import axios from 'axios'
+
+import authorization from '../utils/authorizationMiddleware'
+
+jest.mock('axios')
+const mockedAxios = axios as jest.Mocked<typeof axios>
+
+describe('Authorization Middleware', () => {
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it('throw 403', async () => {
+    const ctx = {
+      status: null,
+      headers: { authorization: null },
+      throw: jest.fn(),
+    }
+
+    mockedAxios.post.mockResolvedValue({ status: 400 })
+
+    await authorization(ctx as never, jest.fn())
```

Comment:
A leitura que estou fazendo aqui é que esse cast para `never` passa a impressão de que `ctx` nunca vai ser usado no método `authorization`. Será que não fica melhor trocar para `unknown`? (o mesmo se aplicar a outros usos de `never` abaixo)

Reply:
O linter do Jetbrains não curtiu muito `any` e `unknown`. Sugeriu usar o `never`. Pessoalmente, eu nem sabia da existência do `never` já que minha experiência codando com typescript antes da VTEX é pouca. Nesse caso teria algum problema ficar assim?

Reply:
Meu entendimento é que semanticamente [never indica algo que nunca deveria existir ou ser chamado](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-never-type)

Podemos ver no pareamento se a gente consegue achar outro jeito de fazer